askama_shared/src/lib.rs

```diff
@@ -294,29 +293,26 @@ static DEFAULT_ESCAPERS: &[(&[&str], &str)] = &[
 ];
 
 #[derive(Debug)]
-pub enum CompileError {
-    Static(&'static str),
-    String(String),
-}
+pub struct CompileError(Cow<'static, str>);
 
 impl fmt::Display for CompileError {
+    #[inline]
     fn fmt(&self, fmt: &mut fmt::Formatter<'_>) -> fmt::Result {
-        match self {
-            CompileError::Static(s) => write!(fmt, "{}", s),
-            CompileError::String(s) => write!(fmt, "{}", s),
-        }
+        fmt::Display::fmt(&self.0, fmt)
     }
 }
 
 impl From<&'static str> for CompileError {
+    #[inline]
     fn from(s: &'static str) -> Self {
-        CompileError::Static(s)
+        CompileError(s.into())
```

Comment:
```suggestion
        Self(s.into())
```

Odd, that this case doesn't trigger the lint. Anyways, might be good to preemptively fix that as well :)